tests/main/base-migration/task.yaml

```diff
@@ -0,0 +1,44 @@
+summary: a snap migrates from base "core" to "core18"
+prepare: |
+  snap install core18
+  snap pack test-snapd-core-migration.base-core
+  snap pack test-snapd-core-migration.base-core18
+execute: |
+  # When we install a snap that is using (implicitly) base: core, then that
+  # snap runs on top of the core16 runtime environment. This can be seen by
+  # looking at the os-release file which will match that of ubuntu core 16.
+  snap install --dangerous test-snapd-core-migration_1_all.snap
+  test-snapd-core-migration.sh -c "cat /usr/lib/os-release" | MATCH 'VERSION_ID="16"'
```

Comment:
Should we also look at the created /run/snapd/ns/*.ifno files here? I.e. that the base_snap_name content is correct?

Reply:
Yeah, good idea. I'll change that soon.

Reply:
Done now